enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,536 @@
+It is primarily used with SR-IOV and as such out of scope for this enhancement mentioned only for completeness.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Not applicable
+
+#### Standalone Clusters
+
+Not applicable
+
+#### Single-node Deployments or MicroShift
+
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints [optional]
+
+First, it must be noted that Multus CNI itself is a meta-CNI. From a high level perspective, its
+purpose is to call other CNIs according to the CNI configs supplied by the user in form of
+NetworkAttachmentDefinitions and Pod annotations.
+Any specific actions are related to the delegate CNIs. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately the responsibility
+for correctness of the configuration is up to the user.
+
+#### Manifests
+
+MicroShift will provide Multus based on thin architecture because of the resource consumption
+(see alternatives for more information).
+Because of the differences with OpenShift (which uses thick Multus plugin), existing OpenShift
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
+
+Because one of the required IPAMs is the `dhcp` (dynamic), manifests will also include
+a DHCP server DaemonSet.
+
+Updating necessary image references will be part of existing rebase procedure.
+
+#### RPM package
+
+RPM spec to build `microshift-multus` and `microshift-multus-release-info` RPMs will be part of
+existing `microshift.spec` file.
+The RPM will include:
+- manifests required to deploy Multus on MicroShift
+- CRI-O drop-in config to use Multus instead of ovn-kubernetes (which will require reorganization
+  of currently existing MicroShift's CRI-O configs)
+- greenboot healthcheck script
+- cleanup script plugin
+
+#### Container images: Multus and network plugins
+
+Because the Multus image used by OpenShift has a large size that is not acceptable for edge devices, new image
+will be prepared and will only include relevant artifacts such as entrypoint script/binary and
+Multus CNI binary (which is copied to host's `/opt/cni/bin`).
+
+To supply network plugins (CNIs) such as `bridge`, `ipvlan`, and `macvlan` a new image will be prepared
+so MicroShift uses the same binaries as OpenShift (alternative is using RHEL's networkplugins RPM - see alternatives).
+This image will also contain IPAM binaries such as `static`, `dynamic (DHCP)`, and `host-local`.
+
+Both of these images will be part of the OpenShift payload which MicroShift references during rebase procedure.
+
+### Risks and Mitigations
```

Comment:
I think it might be possible to create a NAD that impacts the microshift host's networking, e.g.

```
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: troublemaker-nad
spec:
  config: '{
      "cniVersion": "0.3.1",
      "type": "macvlan", 
      "master": "enp1s0",  # Shares the host's main interface
      "ipam": {
          "type": "host-local",
          "subnet": "10.42.0.0/24",  # Overlaps with pod network range
          "routes": [
              {"dst": "0.0.0.0/0", "gw": "10.42.0.100/24"}  # Arbitrary default route
          ] 
      }
   }'
```
If so, this risk presumedly also exists for standard openshift cluster. To what extent are we going to rely on Openshift's multus documentation?